cirq/contrib/acquaintance/executor_test.py

```diff
@@ -0,0 +1,176 @@
+
+    with pytest.raises(ValueError):
+        executor(cirq.Circuit())
+
+    with pytest.raises(TypeError):
+        bad_strategy = AcquaintanceStrategy(cirq.Circuit())
+        bad_strategy.append(cirq.X(qubits[0]))
+        executor(bad_strategy)
+
+    executor(circuit)
+    actual_text_diagram = circuit.to_text_diagram().strip()
+    expected_text_diagram = """
+0: ───0───1───╲0╱─────────────────1───3───╲0╱─────────────────3───5───╲0╱─────────────────5───7───╲0╱─────────────────
+      │   │   │                   │   │   │                   │   │   │                   │   │   │
+1: ───1───0───╱1╲───0───3───╲0╱───3───1───╱1╲───1───5───╲0╱───5───3───╱1╲───3───7───╲0╱───7───5───╱1╲───5───6───╲0╱───
+                    │   │   │                   │   │   │                   │   │   │                   │   │   │
+2: ───2───3───╲0╱───3───0───╱1╲───0───5───╲0╱───5───1───╱1╲───1───7───╲0╱───7───3───╱1╲───3───6───╲0╱───6───5───╱1╲───
+      │   │   │                   │   │   │                   │   │   │                   │   │   │
+3: ───3───2───╱1╲───2───5───╲0╱───5───0───╱1╲───0───7───╲0╱───7───1───╱1╲───1───6───╲0╱───6───3───╱1╲───3───4───╲0╱───
+                    │   │   │                   │   │   │                   │   │   │                   │   │   │
+4: ───4───5───╲0╱───5───2───╱1╲───2───7───╲0╱───7───0───╱1╲───0───6───╲0╱───6───1───╱1╲───1───4───╲0╱───4───3───╱1╲───
+      │   │   │                   │   │   │                   │   │   │                   │   │   │
+5: ───5───4───╱1╲───4───7───╲0╱───7───2───╱1╲───2───6───╲0╱───6───0───╱1╲───0───4───╲0╱───4───1───╱1╲───1───2───╲0╱───
+                    │   │   │                   │   │   │                   │   │   │                   │   │   │
+6: ───6───7───╲0╱───7───4───╱1╲───4───6───╲0╱───6───2───╱1╲───2───4───╲0╱───4───0───╱1╲───0───2───╲0╱───2───1───╱1╲───
+      │   │   │                   │   │   │                   │   │   │                   │   │   │
+7: ───7───6───╱1╲─────────────────6───4───╱1╲─────────────────4───2───╱1╲─────────────────2───0───╱1╲─────────────────
+    """.strip()
+    assert actual_text_diagram == expected_text_diagram
+
+class DiagonalGate(cirq.Gate,
+                   cirq.KnownMatrix,
+                   cirq.TextDiagrammable):
+    def __init__(self, n_qubits: int, diagonal: np.ndarray) -> None:
+        dimension = 2 ** n_qubits
+        if ((diagonal.shape != (dimension,)) or not
+            np.allclose(
+                np.absolute(diagonal), np.ones(dimension))):
+            raise ValueError('Diagonal must be an (2**n_qubits)-dimensional '
+                    'vector with unit-norm entries.')
+        self.n_qubits = n_qubits
+        self.diagonal = diagonal
+
+    def matrix(self) -> np.ndarray:
+        return np.diag(self.diagonal)
+
+    def text_diagram_info(self, args: cirq.TextDiagramInfoArgs
+                          ) -> cirq.TextDiagramInfo:
+        if args.known_qubit_count is None:
+            return NotImplemented
```

Comment:
This is incorrect. `NotImplemented` only makes sense for built-in operators. Just default to a qubit count of 1, or else `raise NotImplementedError()`.

Reply:
Wouldn't `ValueError' make more sense here? `NotImplementedError` seems to imply that we *could* implement it.

Reply:
Yes, that's right. Although I'd go with the "default to 1" approach.